src/lib/y2partitioner/dialogs/lvm_vg.rb

```diff
@@ -0,0 +1,267 @@
+require "yast"
+require "cwm/dialog"
+require "cwm/common_widgets"
+require "y2partitioner/widgets/devices_selection"
+
+Yast.import "Popup"
+
+module Y2Partitioner
+  module Dialogs
+    # Form to set the name, extent size and devices of a volume group
+    class LvmVg < CWM::Dialog
+      # Constructor
+      #
+      # @param controller [Actions::Controllers::LvmVg]
+      def initialize(controller)
+        textdomain "storage"
+        @controller = controller
+      end
+
+      # @macro seeDialog
+      def title
+        _("Add Volume Group")
+      end
+
+      # @macro seeDialog
+      def contents
+        VBox(
+          Left(HVSquash(NameWidget.new(controller))),
+          Left(HVSquash(ExtentSizeWidget.new(controller))),
+          DevicesWidget.new(controller)
+        )
+      end
+
+    private
+
+      # @return [Actions::Controllers::LvmVg]
+      attr_reader :controller
+
+      # Widget for the LVM volume group name
+      class NameWidget < CWM::InputField
+        # Constructor
+        #
+        # @param controller [Actions::Controllers::LvmVg]
+        def initialize(controller)
+          textdomain "storage"
+          @controller = controller
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:notify]
+        end
+
+        # @macro seeAbstractWidget
+        def label
+          # TRANSLATORS: field to enter the name of a new volume group
+          _("&Volume Group Name")
+        end
+
+        # @macro seeAbstractWidget
+        def init
+          self.value = controller.vg_name
+          focus
+        end
+
+        # @macro seeAbstractWidget
+        def handle
+          controller.vg_name = value
+          nil
+        end
+
+        # @macro seeAbstractWidget
+        # Checks whether the given value is valid
+        #
+        # @note An error popup is shown when necessary.
+        #
+        # @return [Boolean]
+        def validate
+          focus
+          presence_validation && legal_characters_validation && uniqueness_validation
+        end
+
+      private
+
+        # @return [Actions::Controllers::LvmVg]
+        attr_reader :controller
+
+        # Sets the focus into this widget
+        def focus
+          Yast::UI.SetFocus(Id(widget_id))
+        end
+
+        # Checks whether some value was entered
+        #
+        # @return [Boolean]
+        def presence_validation
+          return true unless controller.empty_vg_name?
+
+          Yast::Popup.Error(
```

Comment:
NP: In general, having the detection of errors in the controller and the strings explaining them in the widgets feels wrong. We are forced to keep them in sync. We can live like that for the time being (deadlines are around the corner), but maybe some FIXMEs wouldn't hurt.

Reply:
I agree, so the messages generation has been moved to the controller.